components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/IdentityDataStoreServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+package org.wso2.carbon.identity.governance.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.model.IdentityErrorMsgContext;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.governance.model.UserIdentityClaim;
+import org.wso2.carbon.identity.governance.store.JDBCIdentityDataStore;
+import org.wso2.carbon.identity.governance.store.UserIdentityDataStore;
+import org.wso2.carbon.identity.governance.store.UserStoreBasedIdentityDataStore;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.model.ExpressionCondition;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of the IdentityDataStoreService.
+ */
+public class IdentityDataStoreServiceImpl implements IdentityDataStoreService {
+
+    private UserIdentityDataStore identityDataStore;
+
+    private static final String PRE_SET_USER_CLAIM_VALUES = "PreSetUserClaimValues";
+    private static final String PRE_USER_ADD_CLAIM_VALUES = "PreAddUserClaimValues";
+    private static final String STORE_IDENTITY_CLAIMS = "StoreIdentityClaims";
+    private static final String IDENTITY_DATA_STORE_TYPE = "IdentityDataStore.DataStoreType";
+    private static final String DEFAULT_JDBC_IDENTITY_DATA_STORE = "org.wso2.carbon.identity.governance.store" +
+            ".JDBCIdentityDataStore";
+    private static final String DEFAULT_USER_STORE_BASED_IDENTITY_DATA_STORE = "org.wso2.carbon.identity.governance" +
+            ".store.UserStoreBasedIdentityDataStore";
+
+    public IdentityDataStoreServiceImpl() throws ClassNotFoundException, InstantiationException,
+            IllegalAccessException {
+
+        String storeClassName = IdentityUtil.getProperty(IDENTITY_DATA_STORE_TYPE);
+        if (DEFAULT_JDBC_IDENTITY_DATA_STORE.equals(storeClassName)) {
+            identityDataStore =  new JDBCIdentityDataStore();
+        } else if (DEFAULT_USER_STORE_BASED_IDENTITY_DATA_STORE.equals(storeClassName)) {
+            identityDataStore = new UserStoreBasedIdentityDataStore();
+        } else {
+            Class clazz = Class.forName(storeClassName.trim());
+            identityDataStore = (UserIdentityDataStore) clazz.newInstance();
+        }
+    }
+
+    @Override
+    public boolean storeInIdentityDataStore(String userName, UserStoreManager userStoreManager, String operationType,
+                                            Map<String, String> claims) throws UserStoreException {
+
+        // Check if the user is locked.
+        if (PRE_SET_USER_CLAIM_VALUES.equals(operationType)) {
```

Comment:
What is the purpose of this "account lock" check if the rest of the method body does not impact?

Reply:
This is to denote that the account will be locked by storing these identity claims. This was the previous behaviour as well.